lib/src/shared/test/java/com/couchbase/lite/LogTest.java

```diff
@@ -478,6 +464,85 @@ public void testNonASCII() throws CouchbaseLiteException {
         assertTrue(customLogger.getContent().contains("[{\"hebrew\":\"" + hebrew + "\"}]"));
     }
 
+    @Test
+    public void testLogStandardErrorWithFormatting() {
+        Map<String, String> stdErr = new HashMap<>();
+        stdErr.put("FOO", "TEST DEBUG %2$s %1$d %3$.2f");
+        try {
+            Log.initLogging(stdErr);
+
+            BasicLogger logger = new BasicLogger();
+            Database.log.setCustom(logger);
+
+            Log.d(LogDomain.DATABASE, "FOO", new Exception("whoops"), 1, "arg", 3.0F);
+            assertEquals("TEST DEBUG arg 1 3.00 (java.lang.Exception: whoops)", logger.getMessage());
+        }
+        finally {
+            reloadStandardErrorMessages();
+        }
+    }
+
+    @Test
+    public void testLookupStandardMessage() {
+        Map<String, String> stdErr = new HashMap<>();
+        stdErr.put("FOO", "TEST DEBUG");
+        try {
+            Log.initLogging(stdErr);
+            assertEquals("TEST DEBUG", Log.lookupStandardMessage("FOO"));
+        }
+        finally {
+            reloadStandardErrorMessages();
+        }
+    }
+
+    @Test
+    public void testFormatStandardMessage() {
+        Map<String, String> stdErr = new HashMap<>();
+        stdErr.put("FOO", "TEST DEBUG %2$s %1$d %3$.2f");
+        try {
+            Log.initLogging(stdErr);
+            assertEquals("TEST DEBUG arg 1 3.00", Log.formatStandardMessage("FOO", 1, "arg", 3.0F));
+        }
+        finally {
+            reloadStandardErrorMessages();
```

Comment:
Why do we need to reload the error messages after finishing the test? Doesn't the error message happen in CouchbaseLite.init()?

Reply:
Yes it does.  But CouchbaseLite.init happens only once per application.